src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,233 @@
+        if (!car.getDrivers().isEmpty()) {
+            insertCarsDriversRecords(car);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT cars.id, cars.model, cars.manufacturer_id, m.name, m.country "
+                + "FROM cars INNER JOIN manufacturers m ON cars.manufacturer_id = m.id "
+                + "INNER JOIN cars_drivers cd ON cd.car_id = cars.id "
+                + "WHERE cd.driver_id = ?";
+        List<Car> carList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(parseCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car list by driver id "
+                    + driverId, e);
+        }
+        injectDriversInEachCar(carList);
+        return carList;
+    }
+
+    private Car parseCarFromResultSet(ResultSet resultSet) throws SQLException {
+        String model = resultSet.getObject("model", String.class);
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getObject("name", String.class);
+        String manufacturerCountry = resultSet.getObject("country", String.class);
+        Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+        Car car = new Car(model, manufacturer);
+        manufacturer.setId(manufacturerId);
+        car.setId(resultSet.getObject("cars.id", Long.class));
+        return car;
+    }
+
+    private void deleteCarsDriversRecords(Car car) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete from `cars_drivers` by "
+                    + car.getId(), e);
+        }
+    }
+
+    private void insertCarsDriversRecords(Car car) {
```

Comment:
```suggestion
    private void addDriversForCar(Car car) {
```

Reply:
Not agree. I think my version of naming more clearly describes method work